### PR TITLE
Mock and MagickMock subclassing NonCallableMock

### PR DESCRIPTION
**Pull request by rodrigondec**
I forgot to subclass Mock and MagicMock on my quest to have the best stubs on PR  #3846

@srittau

### CODE REVIEW

File: stdlib/3/unittest/mock.pyi [hunk 1]
```diff
@@ -158,8 +159,12 @@ patch: _patcher
 class MagicMixin:
     def __init__(self, *args: Any, **kw: Any) -> None: ...
 
-NonCallableMagicMock = Any
-MagicMock = Any
+class NonCallableMagicMock(MagicMixin, NonCallableMock):
+    pass
```
Comment (srittau) [inline stdlib/3/unittest/mock.pyi]: This has a method:

```python
    def mock_add_spec(self, spec, spec_set=False):
```

File: stdlib/3/unittest/mock.pyi [hunk 2]
```diff
@@ -158,8 +159,12 @@ patch: _patcher
...
+
+class MagicMock(MagicMixin, Mock):
+    pass
```
Comment (srittau) [inline stdlib/3/unittest/mock.pyi]: Same method here.

File: stdlib/3/unittest/mock.pyi (hunk 1 above)
Comment (rodrigondec) [reply stdlib/3/unittest/mock.pyi]: Done

File: stdlib/3/unittest/mock.pyi (hunk 2 above)
Comment (rodrigondec) [reply stdlib/3/unittest/mock.pyi]: Done

**Pull request merged**